File: Project2/Belladonna.cpp
#include "Belladonna.h"
#include "World.h"
#include <stdlib.h>

Belladonna::Belladonna(const Vector2& pos) : Plant(pos, 99, 10, '?')
{
}

Belladonna::~Belladonna()
{
}

void Belladonna::Hit(Organism* attacker)
{
	Plant::Hit(attacker);
	this->Die(attacker);
}

Belladonna* Belladonna::Clone(const Vector2& pos) const
{
	return new Belladonna(pos);
}

std::string Belladonna::GetName() const
{
	return "Belladonna";
}

File: Project2/WorldSimulator/Plants/Belladonna.java
package WorldSimulator.Plants;

import WorldSimulator.Organism;
import WorldSimulator.Plant;
import WorldSimulator.Vector2;

import java.awt.*;

public class Belladonna extends Plant
{
    public Belladonna(Vector2 pos)
    {
        super(pos, 99, 10, '?', new Color(150, 15, 149));
    }

    @Override
    protected void Hit(Organism attacker)
    {
        super.Hit(attacker);
        this.Die(attacker);
    }

    @Override
    public String toString()
    {
        return "Belladonna";
    }
}
